main.cpp

```diff
@@ -19,13 +19,26 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
-
+    CastCallBack(Rewriter& rewriter): _rewriter(rewriter) {};
     void run(const MatchFinder::MatchResult &Result) override {

```

Comment:
> How to prevent parentheses from being added if they are already present in the (int)(f) code?
Для предотвращения возникновения нежеланных скобок можно использовать в нашем исходном выражении isa<ParenExpr> для проверки того, заключено ли оно в скобки:
`if (!isa<ParenExpr>(c_e->getSubExprAsWritten()->IgnoreImpCasts())) 
    newText.append("(");
auto n_e = Lexer::getLocForEndOfToken(expr->getEndLoc(), 0, s_m, LangOptions());
if (!isa<ParenExpr>(c_e->getSubExprAsWritten()->IgnoreImpCasts())) 
    _rewriter.InsertText(endLoc, ")");`

> How to prevent changing the C style cast code for a special case - when we want to suppress warnings about unused variables?
Для этого нужно поместить переменную в выражение (void), чтобы компилятор понимал, что она используется.
Например, можно написать macro 
`#define UNUSED(x) (void)(x)`
Вот такой пример использования
`void f(int x) {
    UNUSED(x);
    ...
}`

> How to prevent changing the C style cast code in macros?
Для предотвращения изменения кода в макросах нужно добавляем проверку:
`if (c_e->getExprLoc().isMacroID()) 
    return;`

Reply:
As for the question:
"How to prevent changing the C style cast code for a special case - when we want to suppress warnings about unused variables?"
The question is how to update your solution to not change the C style cast to the C++ style cast for cases like this: `(void)variable;`?